anaconda.py

```diff
@@ -395,12 +395,21 @@ def _earlyExceptionHandler(ty, value, traceback):
     kspath = startup_utils.find_kickstart(opts)
     log.info("Found a kickstart file: %s", kspath)
 
-    # Run %pre scripts.
-    startup_utils.run_pre_scripts(kspath)
 
     # Parse the kickstart file.
     ksdata = startup_utils.parse_kickstart(kspath, strict_mode=opts.ksstrict)
 
+    from pyanaconda.modules.common.constants.objects import SCRIPTS
+    from pyanaconda.modules.common.constants.services import RUNTIME
+    from pykickstart.constants import KS_SCRIPT_PRE
+    from pyanaconda.modules.common.task import sync_run_task
+
+    # Run %pre scripts.
```

Comment:
IIRC this is why I found handling %pre tricky: running %pre scripts after parse_kickstart seems to be breaking this: https://github.com/rhinstaller/anaconda/blob/1ea55e9d70e49c019a1d8ac4f5e8b46b2f826082/pyanaconda/kickstart.py#L302
So I think for example this case: https://github.com/rvykydal/kickstart-tests/blob/master/bond2-pre.ks.in